### PR TITLE
Exit tile

### LINKED ISSUE #53: Exit tile
- [x] New element, exit tile, that when stepped on triggers an event
- [x] New event for the exit tile
- [x] New view in the UI for the level being completed that opens when the exit tile is stepped on

### PR DESCRIPTION
**Pull request by kieranmillar**
- New floor tile, Exit tile, that ends the level when a protagonist is on one at the end of the turn
- New Level Complete view
- New Level Complete event

resolves #53

### CODE REVIEW

File: src/GameLogic/Enums/TurnEventType.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ export enum TurnEventType {
 	TurnExecuted = 20, // data: undefined
 	LevelLoaded = 21, // data: undefined
 	TileChanged = 22, // data: [number, number] (x and y coordinate)
+	LevelComplete = 23 // data: undefined
```
Comment (EvidentlyCube) [inline src/GameLogic/Enums/TurnEventType.ts]: I think we should put the protagonist who exited in the data for the event so we can create some nice visual effect.

File: src/GameLogic/TurnRunner.ts [hunk 2]
```diff
@@ -29,6 +30,17 @@ export class TurnRunner {
 		level.entities.getEntitiesOfType(EntityType.Fireball).forEach(entity => entity.update(turnState));
 		level.entities.getEntitiesOfType(EntityType.Iceblock).forEach(entity => entity.update(turnState));
 
+		for (let i = 0; i < level.width; i++) {
```
Comment (EvidentlyCube) [inline src/GameLogic/TurnRunner.ts]: I have conflicting feelings about this. I know I don't want to keep it like that in the long run, but I am now questioning whether we should make the exit an entity rather than a floor tile. Thinking back to how a similar issue is solved in DROD - the spikes. They're tracked in an array so that when the game wants to do the spike attack it just has to iterate through that one array rather than the whole level.
And I think that's how we should approach it.
Can you add TODO here referencing #69?

File: src/Scenes/GameScene/Views/ViewLevelComplete.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	private readonly _overlay: PIXI.Sprite;
+
+	private readonly _levelCompleteText: PIXI.BitmapText;
+
+	constructor(stateMachine: GameViewManager) {
+		super();
+
+		this._stateMachine = stateMachine;
+		this._overlay = new PIXI.Sprite(PIXI.Texture.WHITE);
+		this._levelCompleteText = new PIXI.BitmapText('LEVEL COMPLETE!', {
```
Comment (EvidentlyCube) [inline src/Scenes/GameScene/Views/ViewLevelComplete.ts]: Please add a TODO comment referencing #70 here

File: test/GameLogic/e2e/Events/LevelComplete.test.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+	it('Level Complete event should not be raised when a protagonist steps onto an exit tile but also dies', () => {
+		TestLevelBuilder.newLevel()
+			.plotFloor(10, 9, FloorType.Exit)
+			.addEntity(new Fireball(Direction8.Down), 10, 8)
+			.run(PlayerAction.MoveUp)
+			.assertEventNotRaised(TurnEventType.LevelComplete);
+	});
+
+	it('Level Complete event should be raised when a protagonist in an iceblock melts on an exit tile', () => {
```
Comment (EvidentlyCube) [inline test/GameLogic/e2e/Events/LevelComplete.test.ts]: Nice edge case!
